api/src/test/java/org/openmrs/api/OrderServiceTest.java

```diff
@@ -3794,12 +3800,139 @@ public void saveOrderGroup_shouldFailValidationIfAnyOrdersFailValidation() {
+		orderGroupAttributeType.setDatatypeClassname(FreeTextDatatype.class.getName());
+		Context.getOrderService().saveOrderGroupAttributeType(orderGroupAttributeType);
+		assertNotNull(orderGroupAttributeType.getId());
+		assertEquals(initialGroupOrderAttributeTypeCount+1,Context.getOrderService().getOrderGroupAttributeTypes().size());
+	}
+		
+	@Test
+	public void saveOrderGroupAttributeType_shouldEditAnExistingOrderGroupAttributeType(){
+		executeDataSet(ORDER_GROUP_ATTRIBUTES);
+		final String UUID4 ="9cf1bdb2-d18e-11ea-87d0-0242ac130003";
+		final String name ="ECG";
+		//Check for values in the database
+		OrderGroupAttributeType orderGroupAttributeType=Context.getOrderService().getOrderGroupAttributeTypeById(4);
+		assertEquals(UUID4,orderGroupAttributeType.getUuid());
+		assertEquals(name,orderGroupAttributeType.getName());
+		assertEquals("Testing unretire",orderGroupAttributeType.getRetireReason());
+        //edit existing values in the database
+		orderGroupAttributeType.setRetireReason("Change Order To Laparascopy");
+	    orderGroupAttributeType.setName("Laparascopy");
+		orderService.saveOrderGroupAttributeType(orderGroupAttributeType);
+        //confirm new values are persisted
+		assertEquals("Laparascopy",orderGroupAttributeType.getName());
+		assertNotEquals(name,orderGroupAttributeType.getName());
+		assertEquals("Change Order To Laparascopy",orderGroupAttributeType.getRetireReason());
+		assertNotEquals("Testing unretire",orderGroupAttributeType.getRetireReason());
+	}
+			
+	@Test
+	public void retireOrderGroupAttributeType_shouldRetireOrderGroupAttributeType() throws ParseException {
+		executeDataSet(ORDER_GROUP_ATTRIBUTES);
+		OrderGroupAttributeType orderGroupAttributeType = Context.getOrderService().getOrderGroupAttributeTypeById(2);
+		assertFalse(orderGroupAttributeType.getRetired());
+		assertNotNull(orderGroupAttributeType.getRetiredBy());
+		assertNull(orderGroupAttributeType.getRetireReason());
+		assertNull(orderGroupAttributeType.getDateRetired());
+		Context.getOrderService().retireOrderGroupAttributeType(orderGroupAttributeType,"Test Retire");
+		orderGroupAttributeType=Context.getOrderService().getOrderGroupAttributeTypeById(2);
+		assertTrue(orderGroupAttributeType.getRetired());
+		assertNotNull(orderGroupAttributeType.getRetiredBy());
+		assertEquals("Test Retire",orderGroupAttributeType.getRetireReason());
+		assertNotNull(orderGroupAttributeType.getDateRetired(),"True");
+	}
+	
+	@Test
+	public void unretireOrderGroupAttributeType_shouldUnretireOrderGroupAttributeType(){
+		executeDataSet(ORDER_GROUP_ATTRIBUTES);
+		OrderService orderService = Context.getOrderService();
+		OrderGroupAttributeType orderGroupAttributeType = Context.getOrderService().getOrderGroupAttributeTypeById(4);
+        assertTrue(orderGroupAttributeType.getRetired());
+        assertNotNull(orderGroupAttributeType.getRetiredBy());
+        assertNotNull(orderGroupAttributeType.getDateRetired());
```

Comment:
The indention seems off.

Reply:
fixed

Reply:
Are you sure?

Reply:
Yes , but for some unknown reason the changes are not reflecting here .
![DeepinScreenshot_jetbrains-idea-ce_20200908053641](https://user-images.githubusercontent.com/16301386/92427525-624f3280-f195-11ea-8dc4-70ea5d65aca0.png)

Reply:
![IDEA](https://user-images.githubusercontent.com/16301386/92428429-ca9f1380-f197-11ea-9d38-11ae83e2bd89.png)